app/subscriber/src/features/my-reports/edit/content/stories/ContentForm.tsx

```diff
@@ -42,22 +47,101 @@ export const ContentForm: React.FC<IContentFormProps> = ({
   ...rest
 }) => {
   const [{ userInfo }] = useApp();
-  const [{ impersonate }] = useProfileStore();
-
-  if (!content) return null;
-
+  const [{ impersonate, myTonePool }] = useProfileStore();
+  const [{ tonePools }] = useLookup();
   const userId = impersonate?.id ?? userInfo?.id ?? 0;
-  const isAV = content.contentType === ContentTypeName.AudioVideo;
-  const versions = content.versions?.[userId] ?? {
-    byline: content.byline,
-    headline: content.headline,
+  const isAV = content?.contentType === ContentTypeName.AudioVideo;
+  const versions = content?.versions?.[userId] ?? {
+    byline: content?.byline,
+    headline: content?.headline,
     summary: '',
     body: isAV
       ? content.isApproved && content.body
         ? content.body
         : content.summary
-      : content.body,
+      : content?.body,
   };
+  const [initialValues, setInitialValues] = React.useState({
+    tonePools:
+      content?.tonePools && content?.tonePools.length > 0
+        ? content.tonePools.filter((pool: IContentTonePoolModel) => pool.ownerId === userId)
+        : [
+            {
+              ...myTonePool,
+              value:
+                content?.tonePools.find(
+                  (pool: { id: number; ownerId: number }) =>
+                    pool.id === myTonePool?.id && pool.ownerId === userId,
+                )?.value ?? undefined,
+            },
+          ],
+  });
+  const handleSentimentChange = React.useCallback(
+    async (newTonePool: any) => {
+      const validTonePool = Array.isArray(newTonePool) ? newTonePool[0] : null;
+
+      if (!validTonePool) return;
+
+      try {
+        if (!content?.id) {
+          console.error('Content is missing required properties');
+          return;
+        }
+
+        const existingTonePoolIndex = content?.tonePools?.findIndex(
+          (tonePool) => tonePool.ownerId === userId,
+        );
+        let updatedContentTonePool;
+
+        if (existingTonePoolIndex !== undefined && existingTonePoolIndex >= 0) {
+          // Replace the existing TonePool value
+          updatedContentTonePool = content?.tonePools?.map((tonePool, index) =>
+            index === existingTonePoolIndex
+              ? { ...tonePool, value: validTonePool.value }
+              : tonePool,
+          );
+        } else {
+          // Add a new TonePool entry
+          updatedContentTonePool = [
+            ...(content?.tonePools || []),
+            { ...validTonePool, ownerId: userId },
+          ];
+        }
+
+        const updatedContent: IContentModel = {
+          ...content,
+          tonePools: updatedContentTonePool as IContentTonePoolModel[],
+        };
+
+        onContentChange?.(updatedContent);
+      } catch (ex) {
+        console.error('Error updating tone pools:', ex);
+      }
+    },
+    [content, onContentChange, userId],
+  );
+
+  React.useEffect(() => {
+    const updatedTonePools =
+      content?.tonePools && content?.tonePools.length > 0
```

Comment:
Finding the correct tonepool is repeated code.  Make it a utility function to improve maintainability.